instaclustr/resource_gcp_vpc_peering.go

```diff
@@ -0,0 +1,166 @@
+package instaclustr
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceGCPVpcPeering() *schema.Resource {
+	return &schema.Resource{
+		Create: GCPresourceVpcPeeringCreate,
+		Read:   GCPresourceVpcPeeringRead,
+		Update: resourceVpcPeeringUpdate,
+		Delete: resourceVpcPeeringDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceVpcPeeringStateImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"vpc_peering_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"cdc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"peer_vpc_network_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"peer_project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"peer_subnet": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnets"},
+			},
+
+			"peer_subnets": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnet"},
+			},
+		},
+	}
+}
+
+func GCPresourceVpcPeeringCreate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
**[Minor,Desired]** Good to keep the naming of functions consistent. If we are using PascalCase for public visible funcs then it should be `GcpResourceVpcPeeringCreate`, for private funcs it should be camelCase